runtime/lua/vim/lsp/util.lua

```diff
@@ -44,10 +44,20 @@ local function get_border_size(opts)
     height = 2
     width = 2
   else
-    height = height + vim.fn.strdisplaywidth(border[2][1])  -- top
-    height = height + vim.fn.strdisplaywidth(border[6][1])  -- bottom
-    width  = width  + vim.fn.strdisplaywidth(border[4][1])  -- right
-    width  = width  + vim.fn.strdisplaywidth(border[8][1])  -- left
+    local function border_width(id)
+      if type(border[id]) == "table" then
+        -- border specified as a table of <character, highlight group>
+        return vim.fn.strdisplaywidth(border[id][1])
+      elseif type(border[id]) == "string" then
+        -- border specified as a list of border characters
+        return vim.fn.strdisplaywidth(border[id])
+      end
+      error("floating preview border is not correct " .. vim.inspect(border))
```

Comment:
maybe more explicit error message pointing to the help docs and mentioning this should be a type or string. Also border can be "single", "double", "solid" but I think that's handled elsewhere

Reply:
Good point, just added a reference to the docs. The single etc are handled in the code above indeed, so all good.

Reply:
fixed